### PR TITLE
Remove redundant post

### PR DESCRIPTION
**Pull request by thed636**

### CODE REVIEW

File: include/ozo/impl/async_request.h [hunk 1]
```diff
@@ -132,11 +116,12 @@ struct async_send_query_params_op {
...
         }
-        //In the nonblocking state, calls to PQsendQuery, PQputline,
-        //PQputnbytes, PQputCopyData, and PQendcopy will not block
-        //but instead return an error if they need to be called again.
-        while (!send_query_params(conn, query_));
-        post(ctx_, *this);
+
+        if (!send_query_params(conn, query_)) {
+            return done(ctx_, error::pg_send_query_params_failed);
+        };
```
Comment (elsid) [inline include/ozo/impl/async_request.h]: useless semicolon

File: tests/impl/async_request.cpp [hunk 2]
```diff
@@ -50,18 +50,14 @@ TEST_F(async_request_op, should_set_timer_and_send_query_params_and_get_result_a
...
 
     // Get result
-    EXPECT_CALL(executor, post(_)).InSequence(s).WillOnce(InvokeArgument<0>());
-    EXPECT_CALL(strand, dispatch(_)).InSequence(s).WillOnce(InvokeArgument<0>());
     EXPECT_CALL(connection, is_busy()).InSequence(s).WillOnce(Return(false));
     EXPECT_CALL(connection, get_result()).InSequence(s).WillOnce(Return(boost::none));
 
     // Cancel timer
-    EXPECT_CALL(strand, post(_)).InSequence(s).WillOnce(InvokeArgument<0>());
+    // EXPECT_CALL(strand, post(_)).InSequence(s).WillOnce(InvokeArgument<0>());
```
Comment (elsid) [inline tests/impl/async_request.cpp]: just remove this line

File: tests/impl/async_get_result.cpp [hunk 3]
```diff
@@ -363,7 +306,7 @@ TEST_F(async_get_result, should_post_callback_with_error_and_consume_result_if_r
 
     // Post callback with error and cancel all io
     EXPECT_CALL(m.socket, cancel(_)).WillOnce(Return());
-    EXPECT_CALL(m.callback_executor, post(_)).InSequence(s).WillOnce(InvokeArgument<0>());
+    // EXPECT_CALL(m.callback_executor, post(_)).InSequence(s).WillOnce(InvokeArgument<0>());
```
Comment (elsid) [inline tests/impl/async_get_result.cpp]: just remove this line

**Review by elsid — CHANGES_REQUESTED** [review]

File: include/ozo/impl/async_request.h (hunk 1 above)
Comment (thed636) [reply include/ozo/impl/async_request.h]: **Done**

File: tests/impl/async_request.cpp (hunk 2 above)
Comment (thed636) [reply tests/impl/async_request.cpp]: **Done**

File: tests/impl/async_get_result.cpp (hunk 3 above)
Comment (thed636) [reply tests/impl/async_get_result.cpp]: **Done**

**Review by elsid — APPROVED** [review]

**Pull request merged**